### DISCUSSION TITLE
Sequences behaviour

### DISCUSSION THREAD

**Discussion by IsaUlloa** [sngular/kloadgen#275] [Q&A] [answered]
Hi,

While creating the new field "Iterator", we realised that the sequences have the same behaviour as this field when introducing a FieldValueList with more than one value.

We wanted to discuss whether it is ok for the sequences to act like that, or it will be better to remove that behavior from them.

**Comment by jemacineiras** [comment, marked as answer]
I believe it should be removed because it looks like the sequence is behaving like an iteration, when it's not.

**Comment by irodigl** [comment]
I think that in the moment that a iterator has the same behaviour that a sequence, the type of the element automatically change from iterator to sequence and trated like that. So first, I would review the implications for the rest of application before remove it, and if it's possible I would remove it. Thus duplicated code is deleted or refactorized and more clean.

**Reply by IsaUlloa** [reply]
You think this behaviour is more appropriate within the logic of a sequence instead of an iterator. So we should remove it from the iterator, because it is acting like a sequence.

But, as I see it, it is the sequence that acts like an iterator.  Can you explain a little more how do you understand each of these types?

**Reply by irodigl** [reply]
Sorry, I was wrong. I have thinking about it, and I support your idea. It's better mantain one of the options, so I would remove the sequence behaviour and choose the iterator.

**Comment by fcampostato** [comment]
I think having two different operations that arrive at the same behaviour could lead users to think that they or the program are doing something wrong. 
So, at least to me, it seems like it would be better for us as developers and maintainers to eliminate one of the options, and I think the behaviour you're talking about fits better under a definition of an iterator.

**Comment by Rivarsal** [comment]
I believe it should be removed.
